dialog/base.go

```diff
@@ -39,6 +41,22 @@ type dialog struct {
 	parent         fyne.Window
 }
 
+// SetOnClosed allows to set a callback function that is called when
+// the dialog is closed
+func (d *dialog) SetOnClosed(closed func()) {
+	// if there is already a callback set, remember it and call both
+	if d.callback != nil {
```

Comment:
You don’t need a new field here, just set up ‘originalCallback := callback’

Reply:
Ah, sorry, pretty sure I had it that way at first but got me into problems with nil pointers. One shouldn't program to late in the night... Fixed.